client/relay-chain-minimal-node/src/blockchain_rpc_client.rs

```diff
@@ -0,0 +1,490 @@
+		Ok(result)
+	}
+
+	async fn validation_code_hash(
+		&self,
+		at: Hash,
+		para_id: cumulus_primitives_core::ParaId,
+		assumption: polkadot_primitives::v2::OccupiedCoreAssumption,
+	) -> Result<Option<polkadot_primitives::v2::ValidationCodeHash>, sp_api::ApiError> {
+		let result = self
+			.rpc_client
+			.parachain_host_validation_code_hash(at, para_id, assumption)
+			.await?;
+		Ok(result)
+	}
+
+	async fn current_epoch(&self, at: Hash) -> Result<sp_consensus_babe::Epoch, sp_api::ApiError> {
+		let result = self.rpc_client.babe_api_current_epoch(at).await?;
+		Ok(result)
+	}
+
+	async fn authorities(
+		&self,
+		at: Hash,
+	) -> std::result::Result<Vec<polkadot_primitives::v2::AuthorityDiscoveryId>, sp_api::ApiError> {
+		let result = self.rpc_client.authority_discovery_authorities(at).await?;
+		Ok(result)
+	}
+
+	async fn api_version_parachain_host(&self, at: Hash) -> Result<Option<u32>, sp_api::ApiError> {
+		let api_id = <dyn polkadot_primitives::runtime_api::ParachainHost<Block>>::ID;
+		let result = self.rpc_client.runtime_version(at).await.map(|v| v.api_version(&api_id))?;
+		Ok(result)
+	}
+
+	async fn disputes(
+		&self,
+		at: Hash,
+	) -> Result<
+		Vec<(
+			polkadot_primitives::v2::SessionIndex,
+			polkadot_primitives::v2::CandidateHash,
+			polkadot_primitives::v2::DisputeState<polkadot_primitives::v2::BlockNumber>,
+		)>,
+		ApiError,
+	> {
+		let result = self.rpc_client.parachain_host_staging_get_disputes(at).await?;
+		Ok(result)
+	}
+}
+
+#[async_trait::async_trait]
+impl AuthorityDiscovery<Block> for BlockChainRpcClient {
+	async fn authorities(
+		&self,
+		at: Hash,
+	) -> std::result::Result<Vec<polkadot_primitives::v2::AuthorityDiscoveryId>, sp_api::ApiError> {
+		let result = self.rpc_client.authority_discovery_authorities(at).await?;
+		Ok(result)
+	}
+}
+
+impl BlockChainRpcClient {
+	pub async fn local_listen_addresses(
+		&self,
+	) -> Result<Vec<MultiaddrWithPeerId>, RelayChainError> {
+		let addresses = self.rpc_client.system_local_listen_addresses().await?;
+		tracing::debug!(target: LOG_TARGET, ?addresses, "Fetched listen address from RPC node.");
+
+		let mut result_vec = Vec::new();
+		for address in addresses {
+			match MultiaddrWithPeerId::from_str(&address) {
+				Ok(addr) => result_vec.push(addr),
+				Err(err) => return Err(RelayChainError::GenericError(err.to_string())),
+			}
+		}
+
+		Ok(result_vec)
+	}
+
+	pub async fn import_notification_stream_async(
+		&self,
+	) -> Pin<Box<dyn Stream<Item = Header> + Send>> {
+		self.rpc_client
+			.get_imported_heads_stream()
+			.await
+			.expect("subscribe_all_heads")
+			.boxed()
+	}
+
+	pub async fn finality_notification_stream_async(
+		&self,
+	) -> Pin<Box<dyn Stream<Item = Header> + Send>> {
+		self.rpc_client
+			.get_finalized_heads_stream()
+			.await
+			.expect("imported_headers_stream")
+			.boxed()
+	}
+}
+
+fn block_local<T>(fut: impl Future<Output = T>) -> T {
+	let tokio_handle = tokio::runtime::Handle::current();
```

Comment:
I think there is not much to win here by using the `SpawnTaskHandle`. The task manager seems to be mostly designed towards longer running worker tasks. We would need to create channels for the return value of the rpc call since `spawn_blocking` returns `()`. Would be fine but the current solution is much simpler.

Reply:
I don't think that we need to overcomplicate this here, as @skunert said.